apps/08jquery-towers-of-hanoi/script.js

```diff
@@ -2,4 +2,77 @@
 
 $(document).ready(function() {
   // Put app logic here
+  var pieceInHand = false;
+  var $block = null;
+
+  //processes the following function when clicking a data-stack
+  $('[data-stack]').click(function() {
+
+    //If pieceInHand is false and the data-stack contains
+    //children, or blocks, run the if statement to detach
+    //the last block on the data-stack
+
+    if (!pieceInHand && $(this).children().length > 0) {
+      //empty the announce-game-won.  In case the user selects an
+      //empty data-stack
+      $('#announce-game-won').empty();
+      //set $block to the last item on the clicked data-stack
+      $block = $(this).children().last();
+      //detach the $block from the data-stack
+      $block.detach();
+      //change the value of pieceInHand to true to skip to the else statement
+      pieceInHand = true;
+
+      //If pieceInHand is false and the data-stack doesn't contain
+      //children, or blocks, run the else if statement to tell the user to
+      //select a different data-stack
+
+    } else if (!pieceInHand && $(this).children().length === 0) {
+      //update the text of the announce-game-won id
+      $('#announce-game-won').text("That one's empty, select a different one.");
+      return;
+
+      //If pieceInHand is true run the else statement
+    } else
+
+    {
+      //if the data-stack doesn't contain children,
```

Comment:
Great descriptions. Probably a little too much, but I can't complain about that 😄